flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/IntermediateResult.java

```diff
@@ -207,6 +210,10 @@ public boolean isForward() {
         return intermediateDataSet.isForward();
     }
 
+    public boolean isEveryConsumerConsumeAllSubPartitions() {
```

Comment:
I am confused my the method name, the implementation appears to relate to broadcasts.

Reply:
Yes, it was indeed confusing before to have both 'broadcast' and 'EveryConsumerConsumeAllSubPartitions' present at the same time. 
Let me explain the reason for introducing this: `Broadcast` used to represent two specific modes for reading and writing. The writing required `single subpartition contains all data`, and the reading corresponded to `consuming the first subpartition (or any single subpartition)`. 

Now, I am completely replacing `broadcast` with `EveryConsumerConsumeAllSubPartitions` and `singleSubpartitionContainsAllData`. If `singleSubpartitionContainsAllData` is true, then the downstream will only read the first subpartition. Otherwise, we will check if `everyConsumerConsumeAllSubPartitions` is true